### PR TITLE
AEROGEAR-1993 [APB] Update provision playbook to check and configure for shared services

### PR DESCRIPTION
**Pull request by vchepeli**
**Describe what this PR does and why we need it**:
[AEROGEAR-1993](https://issues.jboss.org/browse/AEROGEAR-1993)

**Changes proposed in this pull request**
- Add configure-keycloak.yml
- Provision KeyCloak service only if not SHARED_SERVICE
- Configure KeyCloak service if not SHARED_SERVICE
- Configure KeyCloak service if is SHARED_SERVICE
- Provision Postgres service if is SHARED_SERVICE

**Which issue this PR fixes (This will close that issue when PR gets merged)**
fixes AEROGEAR-1993

### CODE REVIEW

File: roles/provision-keycloak-apb/tasks/configure-keycloak.yml [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+  until: keycloak_auth_response.status == 200
+  retries: 600
+  delay: 2
+
+- name: Generate namespace specific realm
+  template:
+    src: keycloak_realm.json.j2
+    dest: /tmp/keycloak_realm.json
+
+- name: "Create {{ KEYCLOAK_REALM }} realm"
```
Comment (maleck13) [inline roles/provision-keycloak-apb/tasks/configure-keycloak.yml]: what happens here if the realm already exists? It is a use case that the user may wish to reuse an existing realm

File: roles/provision-keycloak-apb/tasks/main.yml [hunk 2]
```diff
@@ -1,12 +1,41 @@
...
 - include_tasks: provision-postgres-ephemeral.yml
-  when: POSTGRESQL_EPHEMERAL is defined
+  when: POSTGRESQL_EPHEMERAL is defined or
+        POSTGRESQL_EPHEMERAL
 
 - include_tasks: provision-keycloak.yml
+  when: SHARED_SERVICE is not defined or
+        not SHARED_SERVICE
+
+- name: "Retrieve route to keycloak"
```
Comment (maleck13) [inline roles/provision-keycloak-apb/tasks/main.yml]: The host can be provided during the provision, I am not seeing where we take advantage of that

File: roles/provision-keycloak-apb/tasks/main.yml [hunk 3]
```diff
@@ -1,12 +1,41 @@
...
+- include_tasks: configure-keycloak.yml
+  vars:
+    KEYCLOAK_HOST="{{ keycloak_route.stdout }}"
+    KEYCLOAK_REALM="{{ namespace }}"
+  when: SHARED_HOST is not defined or
+        not SHARED_SERVICE
+
+- include_tasks: configure-keycloak.yml
+  vars:
+    KEYCLOAK_HOST="{{ SHARED_HOST }}"
```
Comment (maleck13) [inline roles/provision-keycloak-apb/tasks/main.yml]: Do these vars get set for the configure task? is that how the above works here?

**Comment by vchepeli** [comment]
ping @philbrookes, @JameelB, @maleck13

File: roles/provision-keycloak-apb/tasks/configure-keycloak.yml [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+    method: GET
+    validate_certs: no
+    body_format: json
+    headers:
+      Authorization: "bearer {{ keycloak_auth_response.json.access_token }}"
+    status_code: [200, 404]
+    return_content: yes
+  register: keycloak_realm_exists
+
+- debug: var=keycloak_realm_exists
```
Comment (philbrookes) [inline roles/provision-keycloak-apb/tasks/configure-keycloak.yml]: This should be removed.

File: apb.yml [hunk 5]
```diff
@@ -1,6 +1,6 @@
 version: 1.0
 name: keycloak-apb
-description: Keycloak - Open Source Identity and Access Management
+description: Keycloak - AEROGEAR-1993
```
Comment (JameelB) [inline apb.yml]: Should probably change back the description to its original content.

Comment (vchepeli) [reply apb.yml]: This is only for debug/validation part

Comment (vchepeli) [reply apb.yml]: Done!

File: roles/provision-keycloak-apb/tasks/configure-keycloak.yml (hunk 4 above)
Comment (vchepeli) [reply roles/provision-keycloak-apb/tasks/configure-keycloak.yml]: Done!

File: roles/provision-keycloak-apb/tasks/main.yml (hunk 3 above)
Comment (vchepeli) [reply roles/provision-keycloak-apb/tasks/main.yml]: Done!

File: roles/provision-keycloak-apb/tasks/configure-keycloak.yml (hunk 1 above)
Comment (vchepeli) [reply roles/provision-keycloak-apb/tasks/configure-keycloak.yml]: Done!

**Pull request merged**